apps/openmw/mwscript/miscextensions.cpp

```diff
@@ -534,54 +536,65 @@ namespace MWScript
                         {
-                            MWWorld::ConstContainerStoreIterator it = invStorePtr->getSlot (slot);
-                            if (it != invStorePtr->end() && ::Misc::StringUtils::ciEqual(it->getCellRef().getRefId(), item))
-                            {
-                                numNotEquipped -= it->getRefData().getCount();
-                            }
+                            numNotEquipped -= it->getRefData().getCount();
                         }
+                    }
 
-                        for (int slot = 0; slot < MWWorld::InventoryStore::Slots && amount > numNotEquipped; ++slot)
+                    for (int slot = 0; slot < MWWorld::InventoryStore::Slots && amount > numNotEquipped; ++slot)
+                    {
+                        MWWorld::ContainerStoreIterator it = store.getSlot (slot);
+                        if (it != store.end() && ::Misc::StringUtils::ciEqual(it->getCellRef().getRefId(), item))
                         {
-                            MWWorld::ContainerStoreIterator it = invStorePtr->getSlot (slot);
-                            if (it != invStorePtr->end() && ::Misc::StringUtils::ciEqual(it->getCellRef().getRefId(), item))
-                            {
-                                int numToRemove = it->getRefData().getCount();
-                                if (numToRemove > amount - numNotEquipped)
-                                {
-                                    numToRemove = amount - numNotEquipped;
-                                }
-                                invStorePtr->unequipItemQuantity(*it, ptr, numToRemove);
-                                numNotEquipped += numToRemove;
-                            }
+                            int numToRemove = std::min(amount - numNotEquipped, it->getRefData().getCount());
+                            store.unequipItemQuantity(*it, ptr, numToRemove);
+                            numNotEquipped += numToRemove;
                         }
                     }
 
                     int toRemove = amount;
-                    MWWorld::ContainerStore& store = ptr.getClass().getContainerStore (ptr);
                     for (MWWorld::ContainerStoreIterator iter (store.begin()); iter!=store.end(); ++iter)
                     {
-                        if (::Misc::StringUtils::ciEqual(iter->getCellRef().getRefId(), item)
-                                && (!invStorePtr || !invStorePtr->isEquipped(*iter)))
+                        if (::Misc::StringUtils::ciEqual(iter->getCellRef().getRefId(), item) && !store.isEquipped(*iter))
                         {
                             int removed = store.remove(*iter, toRemove, ptr);
                             MWWorld::Ptr dropped = MWBase::Environment::get().getWorld()->dropObjectOnGround(ptr, *iter, removed);
-                            dropped.getCellRef().setOwner("");
+                            dropped.getCellRef().setOwner(std::string());
```

Comment:
It is incosistent with other setOwner usages - we use the "" at least in three different places.

Reply:
Ok.